Mathlib/Combinatorics/SimpleGraph/Connectivity/Subgraph.lean

```diff
@@ -183,6 +183,46 @@ lemma toSubgraph_le_induce_support (p : G.Walk u v) :
   convert Subgraph.le_induce_top_verts
   exact p.verts_toSubgraph.symm
 
+theorem toSubgraph_adj_getVert {u v} (w : G.Walk u v) {i : ℕ} (hi : i < w.length) :
+    (w.toSubgraph).Adj (w.getVert i) (w.getVert (i + 1)) := by
+  induction w generalizing i with
+  | nil => cases hi
+  | cons hxy i' ih =>
+    cases i
+    · simp only [Walk.toSubgraph, Walk.getVert_zero, zero_add, cons_getVert_succ, Subgraph.sup_adj,
+      subgraphOfAdj_adj, true_or]
+    · simp only [Walk.toSubgraph, cons_getVert_succ, Subgraph.sup_adj, subgraphOfAdj_adj, Sym2.eq,
+      Sym2.rel_iff', Prod.mk.injEq, Prod.swap_prod_mk]
+      right
+      exact ih (Nat.succ_lt_succ_iff.mp hi)
+
+theorem getVert_adj_exists_of_toSubgraph_adj {u v u' v'} (w : G.Walk u v)
```

Comment:
I'm suspicious of the naming here. Could you either rename or explain?

Reply:
I looked at it again and made it an `iff` statement instead which seems a bit clearer and also clears up the naming issue.